src/ploneintranet/workspace/browser/tiles/sidebar.py

```diff
@@ -127,22 +136,60 @@ def users(self):
     def existing_users(self):
         return self.workspace().existing_users()
 
-    def __call__(self):
+    def execute_batch_function(self):
         form = self.request.form
         ws = self.workspace()
-        user_id = form.get('user_id')
-        if user_id:
-            if not self.can_manage_workspace():
+        user_ids = form.get('user_id')
+        if isinstance(user_ids, basestring):
+            user_ids = user_ids.split(',')
+        batch_function = form.get('batch-function')
+        if user_ids:
+            if not self.can_manage_roster():
                 msg = _(u'You do not have permission to change the workspace '
                         u'policy')
                 raise Unauthorized(msg)
             else:
-                IWorkspace(ws).add_to_team(user=user_id)
-                api.portal.show_message(
-                    _(u'Member added'),
-                    self.request,
-                    'success',
-                )
+                if batch_function == 'add':
+                    for user_id in user_ids:
+                        IWorkspace(ws).add_to_team(user=user_id)
+                    api.portal.show_message(
+                        _(u'Member(s) added'),
+                        self.request,
+                        'success',
+                    )
+                elif batch_function == 'remove':
+                    for user_id in user_ids:
+                        IWorkspace(ws).remove_from_team(user=user_id)
+                    api.portal.show_message(
+                        _(u'Member(s) removed'),
+                        self.request,
+                        'success',
+                    )
+                elif batch_function == 'role':
+                    role = self.request.get('role')
+                    default_role = ws.participant_policy.title()
+                    groups = [default_role]
+                    if role:
+                        groups.append(role)
```

Comment:
I *think* I just broke the ability for policy exceptions in a recent commit (https://github.com/ploneintranet/ploneintranet/commit/06ac676d57546a66e99d389475f3ba2da61dc6c8) (sorry!)

Previously, this only moved uses from the currently policy to the new policy, but I changed it to move *all* members to the new policy, because we were having persistence issues. I now think this move was foolish, so we may need to switch back to the logic where it only moves users who were in the old policy group (see diff for what it was doing originally).

If you want me to try and fix that, please let me know.